### PR TITLE
Fix CI by allowing retry on failure.

### PR DESCRIPTION
**Pull request by Alexhuszagh**

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
bors try --target x86_64-unknown-linux-gnu

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -265,17 +265,23 @@ jobs:
...
         id: build-docker-image
         if: steps.prepare-meta.outputs.has-image
         timeout-minutes: 120
-        run: cargo xtask build-docker-image -v "${TARGET}${SUB:+.$SUB}"
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 120
+          max_attempts: 3
+          retry_on: error
+          command: cargo xtask build-docker-image -v "${TARGET}${SUB:+.$SUB}"
```
Comment (Emilgardis) [inline .github/workflows/ci.yml:278]: I'm not 100% sure I like this, what if it fails for a valid reason? Then we'll retry unconditionally, and there'll be no cache (I think) for the failed steps so it'll triple up on the errors.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -265,17 +265,23 @@ jobs:
...
             ${{ fromJSON(steps.prepare-meta.outputs.labels) }}
+
       - name: Build Docker image
         id: build-docker-image
         if: steps.prepare-meta.outputs.has-image
         timeout-minutes: 120
-        run: cargo xtask build-docker-image -v "${TARGET}${SUB:+.$SUB}"
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 120
```
[the remark this answers is not in the document]
Comment (Alexhuszagh) [reply .github/workflows/ci.yml:275]: I just set it to the same: every example seemed to require it.

**Comment by Alexhuszagh** [comment]
Closing this temporarily since it seems the outages have stopped, and this has a few barriers that need to be fixed prior to merging that would require mostly rewriting this logic.

**Pull request closed without merging**